third_party/move/move-compiler-v2/src/env_pipeline/inliner.rs

```diff
@@ -953,6 +1109,93 @@ impl<'env, 'rewriter> InlinedRewriter<'env, 'rewriter> {
             Some(new_args_expr)
         };
 
+        // Rewrite and instrument spec block for lambda
+        if let Some(spec) = spec_opt {
+            if let ExpData::SpecBlock(_, spec) = spec.as_ref() {
+                let new_body_node = || env.new_node(body_loc.clone(), body_type.clone());
+
+                // generate pattern for lambda body
+                let body_ty_vec = if let Type::Tuple(tys) = body_type.clone() {
+                    tys.clone()
+                } else {
+                    vec![body_type.clone()]
+                };
+                let mut body_pat_vec = vec![];
+                let mut ret_vars = vec![];
+                for (i, ty) in body_ty_vec.iter().enumerate() {
+                    let new_id = env.new_node(body_loc.clone(), ty.clone());
+                    let sym = env
+                        .symbol_pool()
+                        .make(&format!("{}_{}", LAMBDA_TEMP_RESULT, i + 1));
+                    let pat = Pattern::Var(new_id, sym);
+                    body_pat_vec.push(pat);
+                    let new_id_var = env.new_node(body_loc.clone(), ty.clone());
+                    let var = ExpData::LocalVar(new_id_var, sym).into_exp();
+                    ret_vars.push(var);
+                }
+                let has_result = !ret_vars.is_empty();
+                let body_ret_pat = Pattern::Tuple(new_body_node(), body_pat_vec);
+                let ret_exp_opt = if !has_result {
+                    None
+                } else if ret_vars.len() == 1 {
+                    Some(ret_vars[0].clone())
+                } else {
+                    let new_tuple_id = new_body_node();
+                    Some(ExpData::Call(new_tuple_id, Operation::Tuple, ret_vars).into_exp())
+                };
+
+                // Partition requires and ensures into two spec blocks
+                // and rewrite them to assert
+                let mut lambda_spec_rewriter = LambdaSpecRewriter::new(env, false);
```

Comment:
I wonder whether this already done in [`spec_translator`](https://github.com/aptos-labs/aptos-core/blob/7e55bdc03108d10d3e506c6c07e8ac3799150a3e/third_party/move/move-model/src/spec_translator.rs#L315)

Reply:
Currently the lambda expression is expanded as code block during inling so we need to explicitly instrument the spec block. If we later call the lifted function instead of expanding, the spec_translator will take care of it. Added a TODO for later refactor.

Reply:
Refactor the code to call lifted function instead of expanding the lambda expression so that this code is no longer needed.